en/02_Developer_Guides/02_Controllers/01_Introduction.md

```diff
@@ -130,60 +150,115 @@ public function htmlaction()
  */
 public function ajaxaction() 
 {
-    $this->getResponse()->setBody(json_encode([
-        'json' => true
-    ]));
-
-    $this->getResponse()->addHeader("Content-type", "application/json");
+    $this->getResponse()->addHeader('Content-type', 'application/json');
 
-    return $this->getResponse().
+    return json_encode([
+        'json' => true,
+    ]);
 }
 ```
 
 For more information on how a URL gets mapped to an action see the [Routing](routing) documentation.
 
 ## Security
 
-See the [Access Controller](access_control) documentation.
+See the [Access Control](access_control) documentation.
 
 ## Templates
 
-Controllers are automatically rendered with a template that makes their name. Our `TeamsController` would be rendered
-with a `TeamsController.ss` template. Individual actions are rendered in `TeamsController_{actionname}.ss`. 
+The template to use for a given action is determined in the following order:
+
+1. If a template has been explicitly declared for the action in the `templates` property, it will be used.
+2. If a template has been explicitly declared for the "index" action in the `templates` property, it will be used (regardless of what action is being rendered).
+3. If the `template` property has been set at all, its value will be used.
+4. If a template exists with the name of this class or any of its ancestors, suffixed with the name of the action name, it will be used.
+    - e.g. for the `App\Control\TeamController` example, the "player" action would look for `templates/App/Control/TeamController_player.ss` and `templates/SilverStripe/Control/Controller_player.ss`.
```

Comment:
to follow the recommended action name vs relation name
```suggestion
    - e.g. for the `App\Control\TeamController` example, the "showPlayers" action would look for `templates/App/Control/TeamController_showPlayers.ss` and `templates/SilverStripe/Control/Controller_showPlayers.ss`.
```

Reply:
Done